src/simularium/VisData.ts

```diff
@@ -469,6 +471,18 @@ class VisData {
         }
     }
 
+    // Add parsed frames to the cache and save the timestamp of the first frame
+    private addFramesToCache(frames: ParsedBundle): void {
+        Array.prototype.push.apply(this.frameDataCache, frames.frameDataArray);
+        Array.prototype.push.apply(
+            this.frameCache,
+            frames.parsedAgentDataArray
+        );
+        if (this.firstFrameTime === null) {
+            this.firstFrameTime = frames.frameDataArray[0].time;
```

Comment:
I just noticed this- do you want to use `this.frameDataCache[0].time` instead of `frames.frameDataArray[0].time` here?  i.e if this function is always appending to the cache, do you want to use the first item in the cache or the first item of those being appended?  i guess the null check is protecting that anyway.

Reply:
when does this get reset to null? Is visData re-constructed with each new trajectory? I just dont remember

Reply:
It doesn't get re-constructed with each new trajectory. I reset it manually when a trajectory is cleared (ie with the "Clear Trajectory" button in the viewer testbed) or when a trajectory is changed, in lines 271 and 287 in src/controller/index.ts

Reply:
oh ok, I was looking for it in this file. Probably not a big deal, but is there any sort of "reset" function in this file? If not it might make more sent to make a `clearForNewTrajectory` function that both clears the cache and resets the `firstFrameTime`

Reply:
also is `clearFile` called before `changeFile`? If so it seems like you should only need to reset it in one place

Reply:
No, `clearFile` is only used when file is being cleared not being replaced by a new file. But I will make a `clearForNewTrajectory` function, makes a lot of sense!

Reply:
@toloudis I looked into your question. The cache gets cleared at various points during playback, e.g., when stepping through frames (not sure why it's implemented that way, but I think that's for another discussion) so it's not any more stationary than `frames.frameDataArray`. It doesn't matter which I use because of the null check, like you said, so I'll just keep it like this.